internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -94,26 +121,26 @@ func (r *ForeignClusterReconciler) createFederationRequestIfNotExists(clusterID
 
 	localClusterID := r.clusterID.GetClusterID()
 
-	fr, err := foreignDiscoveryClient.FederationRequests().Get(localClusterID, metav1.GetOptions{})
+	pr, err := foreignClient.PeeringRequests().Get(localClusterID, metav1.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
```

Comment:
@palexster I have a problem here. I'm creating peering requests on foreign cluster and I have to create also associated secret. What can I do? Add permission to create secret on foreign cluster or put it inside PeeringRequest and let a mutating webhook to build that secret?

Reply:
I would say that the first is ok for the short term.